src/trader.rs

```diff
@@ -352,37 +352,61 @@ where
 
         // a 1_000scaling factor is applied to keep some of the decimal
         // TODO: this is bad lol
-        let mut scaling_factor = 1_000.0;
-        let mut mid_price = scaling_factor * (best_ask + best_bid) / 2.0;
+        // let mut scaling_factor = 1_000.0;
+        let scaling_factor = 1.0;
+        let mut mid_price = (best_ask + best_bid) / 2.0;
+        // let mut mid_price = scaling_factor * (best_ask + best_bid) / 2.0;
 
         // apply more scaling factor if necessary
-        while mid_price < 1.0 {
-            scaling_factor *= 1_000.0;
-            mid_price *= 1_000.0;
-        }
+        // while mid_price < 1.0 {
+        //     scaling_factor *= 1_000.0;
+        //     mid_price *= 1_000.0;
+        // }
 
         // Calculate spread:
         let difference = scaling_factor * (best_ask - best_bid).abs();
         let fraction = difference / mid_price;
         // max of 50% fee
-        let spread = (fraction * 100.0 * 100.0).clamp(0.0, 5000.0);
-
-        let numer_scaler = market.start.unit_amount().value();
-        let denom_scaler = market.end.unit_amount().value();
-
-        let pos = Position::new(
-            OsRng,
-            market.into_directed_trading_pair(),
-            spread as u32,
-            // p is always the scaling value
-            (scaling_factor as u128 * denom_scaler).into(),
-            // price is expressed in units of asset 2
-            (mid_price as u128 * numer_scaler).into(),
-            reserves,
+        // min of 1% fee
+        let spread = (fraction * 100.0 * 100.0).clamp(100.0, 5000.0);
+
+        let asset1_scaler = market.start.unit_amount().value();
+        let asset2_scaler = market.end.unit_amount().value();
+
+        // Place two sell-side orders, one selling asset 1 and one selling
+        // asset 2 (since we can't place reserves on both sides in this API).
+        let sell_order1 = format!(
+            "{}{}@{}{}/{}bps",
+            reserves.r2 / asset2_scaler.into(),
+            market.end,
+            mid_price,
+            market.start,
+            spread
+        );
+        tracing::info!("parsing sell order: {}", sell_order1);
+        let sell_order1 = SellOrder::parse_str(&sell_order1)?;
+        let sell_pos1 = sell_order1.into_position(OsRng);
+
+        tracing::trace!(?sell_pos1, "opening sell position");
+        tracing::info!(?sell_pos1, "opening sell position");
+        plan = plan.position_open(sell_pos1);
+
+        let sell_order2 = format!(
```

Comment:
Consider making this a buy order, that way you don't have to invert the midprice, and it's clear that osiris opens a buy + sell to quote each side!